src/QMCWaveFunctions/Fermion/DiracMatrixComputeCUDA.hpp

```diff
@@ -297,21 +309,21 @@ class DiracMatrixComputeCUDA : public Resource
 
   /** Batched inversion and calculation of log determinants.
    *  When TMAT is full precision we can use the a_mat and inv_mat directly
+   *  Side effect of this is after this call a_mats contains the LU factorization
+   *  matrix.
    */
-  template<typename TMAT, typename TREAL>
-  inline std::enable_if_t<std::is_same<T_FP, TMAT>::value> mw_invertTranspose(
+  template<typename TMAT>
+  inline std::enable_if_t<std::is_same<VALUE_FP, TMAT>::value> mw_invertTranspose(
       Resource& resource,
-      RefVector<OffloadPinnedMatrix<TMAT>>& a_mats,
-      RefVector<OffloadPinnedMatrix<TMAT>>& inv_a_mats,
-      OffloadPinnedVector<std::complex<TREAL>>& log_values,
+      RefVector<DualMatrix<TMAT>>& a_mats,
+      RefVector<DualMatrix<TMAT>>& inv_a_mats,
+      DualVector<std::complex<FullPrecReal>>& log_values,
       const std::vector<bool>& compute_mask)
   {
-    assert(log_values.size() == a_mats.size());
+    //assert(log_values.size() == a_mats.size());
```

Comment:
Change?

Reply:
yes if there is still a place where this isn't true when interacting with the application that should get fixed.